internal/controller/apiextensions/composite/reconciler.go

```diff
@@ -403,6 +404,17 @@ func (r *Reconciler) Reconcile(ctx context.Context, req reconcile.Request) (reco
 		"name", cr.GetName(),
 	)
 
+	// Check the pause annotation and return if it has the value "true"
+	// after logging, publishing an event and updating the SYNC status condition
+	if meta.IsPaused(cr) {
+		log.Debug("Reconciliation is paused via the pause annotation", "annotation", meta.AnnotationKeyReconciliationPaused, "value", "true")
+		r.record.Event(cr, event.Normal(reasonPaused, "Reconciliation is paused via the pause annotation"))
+		cr.SetConditions(xpv1.ReconcilePaused())
+		// If the pause annotation is removed, we will have a chance to reconcile again and resume
+		// and if status update fails, we will reconcile again to retry to update the status
+		return reconcile.Result{}, errors.Wrap(r.client.Status().Update(ctx, cr), errUpdateStatus)
+	}
+
 	if meta.WasDeleted(cr) {
```

Comment:
What happens when someone wants to delete a resource that has reconciliation paused?   It seems like the resource will be "stuck" until the paused annotation is removed?  I would think deletion should take precedence as the resource may be in the middle of a larger chain or someone may forget that they placed the paused annotation.

Reply:
I think that would be expected behavior. The user tells to stop reconciling towards the desired state and resource being deleted is the desired state. So, what I'd look for in such "stuck" cases is whether we can do a good job of letting them know that's why they're stuck and event + condition pair seems like the best we could do there.

Reply:
I think this has the potential to cause confusion but as long as it's documented it's fine for now.  Thanks.